### PR TITLE
Set/get correct coord.axislabel with axes.set_xlabel for EllipticalFrame plots

### LINKED ISSUE #10435: Crash when changing vmin and vmax interactively in matplotlib
### Description
When interactively changing vmin and vmax of an image shown with imshow, using astropy wcsaxes in the Qt5Agg MPLBACKEND figure editor python exits and an exception is raised. This happens with matplotlib 3.2.1.

### Expected behavior
Successful change of vmin and vmax.

### Actual behavior
Crash of python with backtrace:

```
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/matplotlib/backends/qt_editor/_formlayout.py", line 481, in apply
    self.apply_callback(self.formwidget.get())
  File "/usr/lib/python3.8/site-packages/matplotlib/backends/qt_editor/figureoptions.py", line 194, in apply_callback
    axes.set_xlabel(xlabel)
  File "/home/beischer/repos/astropy/astropy/visualization/wcsaxes/core.py", line 471, in set_xlabel
    raise TypeError("set_xlabel() missing 1 required positional argument: 'xlabel'")
TypeError: set_xlabel() missing 1 required positional argument: 'xlabel'
zsh: abort (core dumped)  MPLBACKEND=Qt5Agg python test_mpl.py
```

### Steps to Reproduce

Consider this script:
```python
import numpy as np
import matplotlib.pyplot as plt
from astropy.visualization.wcsaxes.frame import EllipticalFrame
from astropy.io import fits
from astropy.wcs import WCS
from astropy import units as u

data = np.ones((360,720))

header = fits.Header()
header['BITPIX'] = -32
header['NAXIS']  = 2
header['NAXIS1'] = 720
header['CRPIX1'] = 360.5
header['CRVAL1'] = 0.0
header['CDELT1'] = -0.5
header['CTYPE1'] = 'RA---AIT'
header['CUNIT1'] = 'deg'
header['NAXIS2'] = 360
header['CRPIX2'] = 180.5
header['CRVAL2'] = 0.0
header['CDELT2'] = 0.5
header['CTYPE2'] = 'DEC--AIT'

w = WCS(header)
pix_l_min, _ = w.all_world2pix(179.9, 0.0, 0)
pix_l_max, _ = w.all_world2pix(-179.9, 0.0, 0)
_, pix_b_min = w.all_world2pix(0, -89.9, 0)
_, pix_b_max = w.all_world2pix(0,  89.9, 0)

plt.figure()
plt.subplot(projection = w, frame_class=EllipticalFrame)
im = plt.imshow(data, origin='lower', cmap='jet')
ax = plt.gca()
ax.coords.grid(color='black', ls='dotted')
lon = ax.coords[0]
lat = ax.coords[1]
lon.set_axislabel('Right Ascension', color='black')
lat.set_axislabel('Declination', color='black')
lon.set_major_formatter('d')
lat.set_major_formatter('d')
lon.set_ticklabel(color='black')
lat.set_ticklabel(color='black')
lon.set_ticks(spacing = 60 * u.degree, color='black')
lat.set_ticks(spacing = 30 * u.degree, color='black')
im.set_clip_path(ax.coords.frame.patch)
plt.xlim(pix_l_min, pix_l_max)
plt.ylim(pix_b_min, pix_b_max)
plt.show()
```

1. MPLBACKEND=Qt5Agg python test_mpl.py
2. Click on the 'edit axis' button (z-shaped arrow pointing to the top right)
3. Click on 'Images, etc.'
4. Change 'Min. value'
5. Click 'Apply'

### System Details
```
Python 3.8.3 (default, May 17 2020, 18:15:42) 
[GCC 10.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform; print(platform.platform())
Linux-5.7.0-arch1-1-x86_64-with-glibc2.2.5
>>> import sys; print("Python", sys.version)
Python 3.8.3 (default, May 17 2020, 18:15:42) 
[GCC 10.1.0]
>>> import numpy; print("Numpy", numpy.__version__)
Numpy 1.18.4
>>> import astropy; print("astropy", astropy.__version__)
astropy 4.1rc1
>>> import scipy; print("Scipy", scipy.__version__)
Scipy 1.4.1
>>> import matplotlib; print("Matplotlib", matplotlib.__version__)
Matplotlib 3.2.1
```

### PR DESCRIPTION
**Pull request by dhomeier**
Co-authored-by:  giang-nghg@users.noreply.github.com

### Description
#10435 has exposed a bug where both `axes.xlabel` and `axes.ylabel` remain set to `None` in a 
`(projection = WCS(hdr), frame_class=EllipticalFrame)` plot, causing any function that subsequently tries to access the labels to raise an error.
https://github.com/astropy/astropy/issues/10435#issuecomment-638429656 has identified the underlying reason as `WCSAxes.set_xlabel()` and colleagues only checking for the associated `coord.axislabels` under their `RectangularFrame` labels, not the `EllipticalFrame` ones.

This patch has the functions always checking for both versions of the corresponding label; since the labels are unique and the two frame types above seem to be the only valid `frame_class`s for these plots, this _should_ have all configurations covered. Test is checking the correct label settings for both.
Tested with Matplotlib 3.2.1 on Python 3.7.7 and 3.8.3, and with MPL 2.1.1 on Python 3.6.10.

#### Note
The label references in `get_coords_overlay` also seem to be specific to `RectangularFrame` (`RectangularFrame` plots have `axes.frame_class.spine_names` `'brtl'`, `EllipticalFrame` ones `'chv'`), but I have not looked further into whether they should be extended as well, or how.

**EDIT:** fixes #10435

### CODE REVIEW

**Comment by pllim** [comment]
tl;dr -- So this completely fixes #10435 ? Thanks!

**Comment by dhomeier** [comment]
> tl;dr -- So this completely fixes #10435 ? Thanks!

All the exceptions I could produce, and I believe @bastianbeischer as well.

**Comment by pllim** [comment]
Thanks, @dhomeier ! I have marked it as such.

**Comment by dhomeier** [comment]
Thanks @plim , could not mark it myself or maybe chose the wrong words in the description!

**Comment by astrofrog** [comment]
Just to check, is it correct that this fix can't be backported to v4.0.x? (since it's currently milestoned as 4.1.1)

**Comment by dhomeier** [comment]
> Just to check, is it correct that this fix can't be backported to v4.0.x? (since it's currently milestoned as 4.1.1)

Technically there is no problem; the patch applies cleanly to 4.0.1 and I've tested it locally on Python 3.6.

**Comment by pllim** [comment]
Okay then, I have re-milestoned it to 4.0.2.

**Comment by pllim** [comment]
Thanks, everyone!
